### PR TITLE
feat(radio-group): component

### PR DESCRIPTION
**Pull request by wassim-arfa**

### CODE REVIEW

**Comment by arfa** [comment]
headless ui

https://headlessui.dev/react/radio-group

File: components/lib/Radio/Radio.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      <RadioGroup
+        value={selected}
+        onChange={(value) => {
+          setSelected(value)
+          onChange && onChange(value)
+        }}
+        className="flex items-center space-x-3"
+      >
+        <div className="grid grid-cols-7 gap-2">
+          {attributes?.map(({ id, value, meta }) => (
```
Comment (arfa) [inline components/lib/Radio/Radio.tsx:41]: options

File: components/lib/Radio/Radio.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                )
+              }
+            >
+              {title === 'color' ? (
+                <span
+                  className="block w-full h-full rounded"
+                  style={{ backgroundColor: meta }}
+                />
+              ) : (
+                <RadioGroup.Label as="p"> {value} </RadioGroup.Label>
```
Comment (arfa) [inline components/lib/Radio/Radio.tsx:60]: toujours mettre le label 
color || value => color, value, color + value

File: components/lib/Radio/Radio.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import { useState } from 'react'
+interface Props {
+  className?: string
+  title: string
+  showTitle?: boolean
+  options: {
+    id: number
+    value: string
+    meta: string
+  }[]
```
Comment (arfa) [inline components/lib/Radio/Radio.tsx:12]:
```
{label: 'label', value: 'value', color: '#...'}

File: components/lib/Radio/Radio.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  showTitle = true,
+  showValue = true,
+  showcolor = false,
+  options,
+  onChange,
+}) => {
+  const [selected, setSelected] = useState()
+  return (
+    <div className={className}>
+      {showTitle ? (
```
Comment (arfa) [inline components/lib/Radio/Radio.tsx:30]: title

- showTitle

File: components/lib/Radio/Radio.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+export const Radio: React.FC<Props> = ({
+  className = 'mb-4',
+  title,
+  showTitle = true,
+  showValue = true,
+  showcolor = false,
+  options,
+  onChange,
```
Comment (arfa) [inline components/lib/Radio/Radio.tsx:25]: as = "span"

File: components/lib/Radio/Radio.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+              }
+            >
+              <RadioGroup.Label
+                as={'a'}
+                className={
+                  showcolor
+                    ? '-m-0.5 relative p-0.5 rounded w-full flex h-full items-center justify-center text-gray-50 cursor-pointer focus:outline-none'
+                    : ''
+                }
+                style={showcolor ? { backgroundColor: meta } : {}}
```
Comment (arfa) [inline components/lib/Radio/Radio.tsx:65]:
```jsx
style={{ backgroundColor: color || '' }}

File: components/lib/Radio/Radio.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  {
+    id: 1,
+    value: 'M',
+    meta: 'rgb(212 117 117)',
+  },
+]
+export const Default = Template.bind({})
+
+Default.args = {
+  title: 'size', // try title: 'color'
```
Comment (arfa) [inline components/lib/Radio/Radio.stories.tsx:32]: delete comment

File: components/lib/Radio/Radio.tsx (hunk 6 above)
Comment (wassim-arfa) [reply components/lib/Radio/Radio.tsx]: obligation d'affichage de coulor l'orsqu'il y a un color

File: components/lib/Radio/Radio.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+              <RadioGroup.Label
+                as={'span'}
+                className={
+                  color
+                    ? '-m-0.5 relative p-0.5 rounded w-full flex h-full items-center justify-center text-gray-50 cursor-pointer focus:outline-none'
+                    : ''
+                }
+                style={color ? { backgroundColor: color } : {}}
+              >
+                {label ? label : null}
```
Comment (arfa) [inline components/lib/Radio/Radio.tsx:62]: {label}

**Pull request merged**